lib/rubocop/cop/lint/duplicate_in_pattern.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks that there are no repeated patterns used in `in` keywords.
+      #
+      # @example
```

Comment:
It'd be nice to show examples with more complex patterns, so it's clear that this cop is pretty smart about them.